mlir/lib/Dialect/Vector/Transforms/LowerVectorTranspose.cpp

```diff
@@ -336,6 +336,27 @@ class TransposeOpLowering : public OpRewritePattern<vector::TransposeOp> {
       return rewriter.notifyMatchFailure(
           op, "Options specifies lowering to shuffle");
 
+    // Replace:
+    //   vector.transpose %0, [1, 0] : vector<nx1x<eltty>> to
+    //                                 vector<1xnxelty>
+    // with:
+    //   vector.shape_cast %0 : vector<nx1x<eltty>> to vector<1xnxelty>
+    //
+    // Source with leading unit dim (inverse) is also replaced. Unit dim must
+    // be fixed. Non-unit can be scalable.
+    if (resType.getRank() == 2 &&
+        ((resType.getShape().front() == 1 &&
+          !resType.getScalableDims().front()) ||
+         (resType.getShape().back() == 1 &&
```

Comment:
The test on `resType.getShape/getScalableDims().back()` can be omitted thanks to the semantics of the vector.transpose op.

Reply:
without this the example above wont be replaced with shape_cast, unless I'm missing something?

Reply:
sorry I meant the inverse case, e.g.
```
%0 = vector.transpose %arg0, [1, 0] : vector<1x4xf32> to vector<4x1xf32>
->
%0 = vector.shape_cast %arg0 : vector<1x4xf32> to vector<4x1xf32>
```

Reply:
You're right, for some reason I thought you were looking at the source and the res type and duplicating test.
Sorry for the noise.

Reply:
no worries thanks for reviewing